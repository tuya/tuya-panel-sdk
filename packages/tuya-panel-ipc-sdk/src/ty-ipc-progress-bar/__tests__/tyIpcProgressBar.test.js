import React from 'react';
import { shallow } from 'enzyme';
import TYIpcProgressBar from '../index';

const defaultBarData1 = {
  key: 'example3',
  value: 0,
  min: 0,
  max: 100,
  disabled: true,
  showPercentUnit: true,
  maxColor: 'red',
  minColor: 'green',
}

const defaultBarData2 = {
  key: 'example3',
  value: 0,
  min: 0,
  max: 100,
  disabled: true,
  showPercentUnit: true,
  maxColor: 'red',
  minColor: 'green',
  customUnitText: 'unit',
  noTitle: true,
}

describe('TYIpcProgressBar components', () => {
  it('basic render', () => {
    const wrapper = shallow(
      <TYIpcProgressBar/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('basic render disabled', () => {
    const wrapper = shallow(
      <TYIpcProgressBar barData={defaultBarData1}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('basic render disabled with image', () => {
    const wrapper = shallow(
      <TYIpcProgressBar barData={defaultBarData2}/>);
    expect(wrapper).toMatchSnapshot();
  });

});
