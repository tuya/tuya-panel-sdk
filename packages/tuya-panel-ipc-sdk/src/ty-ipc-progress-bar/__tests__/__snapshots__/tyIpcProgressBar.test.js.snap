// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TYIpcProgressBar components basic render 1`] = `
<View
  style={
    Array [
      Object {
        "height": 112,
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "justifyContent": "flex-end",
            "width": 80,
          },
          Object {},
        ]
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 24,
            },
            Object {},
          ]
        }
        type={null}
        weight={null}
      >
        0
      </TYText>
    </View>
    <ThemedSlider
      disabled={false}
      maximumTrackTintColor="#eee"
      maximumValue={100}
      minimumTrackTintColor="#eee"
      minimumValue={0}
      onSlidingComplete={[Function]}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "marginLeft": 16,
            "marginRight": 10,
          },
          Object {},
        ]
      }
      theme={null}
      value={0}
    />
    <View
      style={
        Array [
          Object {
            "minWidth": 40,
          },
          Object {},
        ]
      }
    >
      <TYText
        align={null}
        color={null}
        numberOfLines={1}
        size={null}
        style={
          Array [
            Object {
              "fontSize": 24,
            },
            Object {
              "color": "#333",
            },
            Object {},
          ]
        }
        type={null}
        weight={null}
      >
        0
      </TYText>
    </View>
  </View>
</View>
`;

exports[`TYIpcProgressBar components basic render disabled 1`] = `
<View
  style={
    Array [
      Object {
        "height": 112,
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "justifyContent": "flex-end",
            "width": 80,
          },
          Object {},
        ]
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Array [
            Object {
              "color": "#aeaeae",
              "fontSize": 24,
            },
            Object {},
          ]
        }
        type={null}
        weight={null}
      >
        0%
      </TYText>
    </View>
    <ThemedSlider
      disabled={true}
      maximumTrackTintColor="#eee"
      maximumValue={100}
      minimumTrackTintColor="#eee"
      minimumValue={0}
      onSlidingComplete={[Function]}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "marginLeft": 16,
            "marginRight": 10,
          },
          Object {},
        ]
      }
      theme={null}
      value={0}
    />
    <View
      style={
        Array [
          Object {
            "minWidth": 80,
          },
          Object {},
        ]
      }
    >
      <TYText
        align={null}
        color={null}
        numberOfLines={1}
        size={null}
        style={
          Array [
            Object {
              "fontSize": 24,
            },
            Object {
              "color": "#aeaeae",
            },
            Object {},
          ]
        }
        type={null}
        weight={null}
      >
        0%
      </TYText>
    </View>
  </View>
</View>
`;

exports[`TYIpcProgressBar components basic render disabled with image 1`] = `
<View
  style={
    Array [
      Object {
        "height": 112,
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {},
      ]
    }
  >
    <ThemedSlider
      disabled={true}
      maximumTrackTintColor="#eee"
      maximumValue={100}
      minimumTrackTintColor="#eee"
      minimumValue={0}
      onSlidingComplete={[Function]}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "marginLeft": 16,
            "marginRight": 10,
          },
          Object {},
        ]
      }
      theme={null}
      value={0}
    />
    <View
      style={
        Array [
          Object {
            "minWidth": 80,
          },
          Object {},
        ]
      }
    >
      <TYText
        align={null}
        color={null}
        numberOfLines={1}
        size={null}
        style={
          Array [
            Object {
              "fontSize": 24,
            },
            Object {
              "color": "#aeaeae",
            },
            Object {},
          ]
        }
        type={null}
        weight={null}
      >
        unit
      </TYText>
    </View>
  </View>
</View>
`;
